docs/index.asciidoc

```diff
@@ -31,6 +31,22 @@ guaranteed!
 
 Beware, this gem does not yet support codecs. Please use the 'format' option for now.
 
+[id="plugins-{type}s-{plugin}-retry_policy"]
+==== Retry policy
+
+This output has two levels of retry, library and plugin.
+
+The library only retry IO related failures. Non retriable IOException are InterruptedIOException, UnknownHostException, ConnectException and SSLException.
```

Comment:
From a user perspective maybe is better not to report the Java exceptions names, but the concept they carries.
So it could be something like:
```
Non retriable errors could regard SSL problems or host not resolved but also connection problem or OS/JVM level interruptions happening during a request.
```